deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/BaseRecurrentLayer.java

```diff
@@ -42,10 +52,33 @@ public void setNIn(InputType inputType, boolean override) {
     public InputPreProcessor getPreProcessorForInputType(InputType inputType) {
         return InputTypeUtil.getPreprocessorForInputTypeRnnLayers(inputType, getLayerName());
     }
-
+    //class Builder<T extends FeedForwardLayer.Builder<T>> extends Layer.Builder<T>
 
     @AllArgsConstructor
-    public static abstract class Builder<T extends Builder<T>> extends FeedForwardLayer.Builder<T> {
+    public static abstract class Builder<T extends BaseRecurrentLayer.Builder<T>> extends FeedForwardLayer.Builder<T> {
+        protected boolean useLayerNormalization = false;
+
+
+        public Builder(Builder<T> builder) {
```

Comment:
You have such constructors in the code base, just not consistently: see BaseRecurrentLayer, BaseOutputLayer. The goal is to construct a builder from another and transfer the configuration to the new instance. Whether this is needed in practice is unclear, but since (1) I saw examples where it was done correctly (see BaseOutputLayer) and (2) I had to because the super class does not have a no arg constructor, I also implemented the functionality. 
Removing the constructor with builder won't compile unless I remove it from the superclass as well (BaseRecurrentLayer).